paasta_tools/autoscaling/max_all_k8s_services.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python
+from paasta_tools.kubernetes.application.controller_wrappers import (
+    get_application_wrapper,
+)
+from paasta_tools.kubernetes_tools import KubeClient
+from paasta_tools.kubernetes_tools import KubernetesDeploymentConfig
+from paasta_tools.paasta_service_config_loader import PaastaServiceConfigLoader
+from paasta_tools.utils import get_services_for_cluster
+from paasta_tools.utils import load_system_paasta_config
+
+
+def main() -> None:
+    system_paasta_config = load_system_paasta_config()
+
+    kube_client = KubeClient()
+
+    services = {
+        service
+        for service, instance in get_services_for_cluster(
+            cluster=system_paasta_config.get_cluster(), instance_type="kubernetes"
+        )
+    }
+
+    for service in services:
+        pscl = PaastaServiceConfigLoader(service=service, load_deployments=False)
+        for instance_config in pscl.instance_configs(
+            cluster=system_paasta_config.get_cluster(),
+            instance_type_class=KubernetesDeploymentConfig,
+        ):
+            # Force min_instances = max_instances so that we scale up.
+            max_instances = instance_config.get_max_instances()
+            if max_instances is not None:
+                instance_config.config_dict["min_instances"] = max_instances
+                formatted_application = instance_config.format_kubernetes_app()
+                wrapper = get_application_wrapper(formatted_application)
+                wrapper.soa_config = instance_config
+                print(f"Scaling up {service}.{instance_config.instance}")
+                wrapper.update(kube_client)
```

Comment:
Since you change min_instances to max_instances, in get_autoscaling_metric_spec, HPA sets min_replicas to max_replicas. And with limit_instance_count function, replicas in formatted_application is also set to max_instances. 

But how do you make sure when setup_kubernetes_job is called, this won't get replaced?

Reply:
For example, here, https://github.com/Yelp/paasta/blob/df9c867fe1bcdf714ba85da84a8f2bb90a0105b6/paasta_tools/kubernetes_tools.py#L389 min_replicas is still min_replicas when run by setup_kubernetes_job

Reply:
Perhaps we should not update min_replicas and max_replicas in HPA when they are the same. And not allow users to set them as the same.

Reply:
Oh, good point, I hadn't thought about that.

Reply:
@EmanekaT in fef79425 I set replicas on the formatted k8s object instead of changing min_instances. I think this should fix this problem.